modules/py_ml.c

```diff
@@ -426,11 +427,159 @@ static MP_DEFINE_CONST_OBJ_TYPE(
     locals_dict, &py_ml_model_locals_dict
     );
 
-extern const mp_obj_type_t py_ml_nms_type;
+// The function finds the maximum value in each row of a 2D ndarray and returns the indices
+// of the rows where the maximum exceeds a threshold which is the most CPU intensive post-processing
+// step. This function can handle regular and transposed ndarrays where the strides result in
+// non-contiguous memory access patterns. TODO: Futher performance optimizations are possible for
+// contiguous int8/uint8/int16/uint16 ndarrays by processing contiguous rows using contiguous
+// memory operations and integer SIMD operations.
+static mp_obj_t py_ml_threshold(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
```

Comment:
How faster is this compared to old one for 10K run? With no benchmarks, how do I know this is faster? Maybe it's the same or slower?